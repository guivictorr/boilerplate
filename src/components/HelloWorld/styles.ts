import styled from 'styled-components';

export const Container = styled.main`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  height: 100vh;
  background: #1a1a1a;
`;

export const Title = styled.h1`
  color: #f5f5f5;
`;
