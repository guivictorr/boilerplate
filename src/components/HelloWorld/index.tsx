import * as S from './styles';

const HelloWorld = () => (
  <S.Container>
    <S.Title>Hello World</S.Title>
  </S.Container>
);

export default HelloWorld;
