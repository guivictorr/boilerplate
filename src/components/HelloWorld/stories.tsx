import HelloWorld from '.';
import { Meta, Story } from '@storybook/react';

export default {
  title: 'Components/HelloWorld',
  component: HelloWorld
} as Meta;

export const Basic: Story<{ title: string }> = args => <HelloWorld {...args} />;
Basic.args = {
  title: 'Hello storybook'
};
