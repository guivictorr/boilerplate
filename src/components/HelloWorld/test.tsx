import { render, screen } from '@testing-library/react';

import HelloWorld from '.';

describe('<HelloWorld />', () => {
  it('should render the heading', () => {
    const { container } = render(<HelloWorld />);

    expect(
      screen.getByRole('heading', { name: /hello world/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
