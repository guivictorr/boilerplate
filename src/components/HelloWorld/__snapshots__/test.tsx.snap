// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelloWorld /> should render the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  background: #1a1a1a;
}

.c1 {
  color: #f5f5f5;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Hello World
  </h1>
</main>
`;
